lib/client/player.go

```diff
@@ -215,7 +238,30 @@ func (p *sessionPlayer) playRange(from, to int) {
 		}
 		// played last event?
 		if i == len(p.sessionEvents) {
-			p.Stop()
+			// we defer here so the stop notification happens after the deferred final state update
+			defer p.stopOnce.Do(func() { close(p.stopC) })
```

Comment:
I'm pretty sure defers are LIFO so wouldn't this still happen first?

Reply:
Yep, not sure what I was thinking here. Fixed.